SetupWebpages.py

```diff
@@ -0,0 +1,86 @@
+#####IMPORTS#####
+import os
+import WebpageFunctions
+import json
+
+# Grab environment variables - wildly inconsistent variable naming. Will fix at some point.
+ValidationPath = os.getenv("ValidationPath")
+TRAVIS_PULL_REQUEST = os.getenv("TRAVIS_PULL_REQUEST","false")
+TRAVIS_PULL_REQUEST_TITLE = os.getenv("TRAVIS_PULL_REQUEST_TITLE","")
+TRAVIS_PULL_REQUEST_LINK = os.getenv("TRAVIS_PULL_REQUEST_LINK","")
+TRAVIS_COMMIT = os.getenv("TRAVIS_COMMIT","")
+GITHUBTOKEN = os.getenv("GITHUBTOKEN","")
+
+
+#Checkout the validation webpage branch - common function, should add as such.
+WebpageFunctions.checkout_validation_webpage_branch(ValidationPath,TRAVIS_PULL_REQUEST,TRAVIS_COMMIT)
+
+
+# Do some specific things here. Can this be made into a common function? No way of testing...
+if TRAVIS_PULL_REQUEST != "false":
+        travis_commit_message = f" Pull Request #{TRAVIS_PULL_REQUEST}: {TRAVIS_PULL_REQUEST_TITLE}"
+
+# First update the list of commits
+with open(f"{ValidationPath}/Webpage/folderlist.new", "w") as f_new:
+    f_new.write("{TRAVIS_COMMIT} \n")
+    with open(f"{ValidationPath}/Webpage/folderlist", "r") as f:
+        f_new.write(f.read())
+os.rename(f"{ValidationPath}/Webpage/folderlist.new", f"{ValidationPath}/Webpage/folderlist")
+
+# Update the main page. This seems a bit silly, is TRAVIS_PULL_REQUEST guaranteed to be an integer If it does not equal false?

```

Comment:
What this test is doing is making the correct clicky link to jump to the relevant github page that initiated the CI

At the moment the logic is
- if it's a PR, create a link to the PR
- if it's not a PR (e.g. from a direct push, or merge, or tag), don't create a link

This is coming from [here](https://github.com/WCSim/WCSim/blob/develop/.github/workflows/c-cpp.yml#L42) which, according to [this](https://stackoverflow.com/a/62436027), which gives a postive integer if it's from a PR

I do have to admit that I can't find what value it takes if it's not a PR (is it `false` or `0` or `"false"` or whatever). So we'll have to test to see if this breaks